### PR TITLE
feat: add props for adding custom arrows to calendar component

### LINKED ISSUE #1102: Calendar - allow props way of restyling of prev/next buttons
## 🚀 Feature Proposal

I would like to have a possibility to pass some prop into Datepicker (further - Calendar), which would then be picked up by Calendar, that would allow me to style header buttons myself, without overwriting ghost buttons on the page.

## Motivation

Calendar is using ghost button to render the prev/next buttons to render header buttons:
https://github.com/akveo/react-native-ui-kitten/blob/master/src/components/ui/calendar/components/calendarHeader.component.tsx#L85
In our application, we are using the Datepicker component, and we had to restyle all the buttons, including the ghost button (medium size) to have a paddingHorizontal of 30px. Unfortunately, Calendar also picked up this change, causing the buttons to overflow to the right when the month had longer name (they didn't fit into the calendar).

I've coped with it by changing the padding on the ghost button to much smaller number, though it might happen we will need to bring back the old padding. In that case, it would be very useful to have some way to overwrite style of those buttons, without overwriting all.

Perhaps you could suggest me any other way this could be achieved without making changes to the kitten component? I would appreciate that greatly!

## Example

Please provide an example for how this feature would be used.
```
<Calendar ... headerButtonsStyle={{ paddingHorizontal: 0 }} />
```

### PR DESCRIPTION
**Pull request by whitestranger7**
### Please read and mark the following check list before creating a pull request:

 - [x] I read and followed the [CONTRIBUTING.md](https://github.com/akveo/react-native-ui-kitten/blob/master/CONTRIBUTING.md) guide.
 - [x] I read and followed the [New Feature Checklist](https://github.com/akveo/react-native-ui-kitten/blob/master/DEV_DOCS.md) guide.

 #### Short description of what this resolves:
Closes #1102

### CODE REVIEW

File: src/components/ui/calendar/components/calendarHeader.component.tsx [hunk 1]
```diff
@@ -82,19 +85,45 @@ export class CalendarHeader extends React.Component<CalendarHeaderProps> {
...
   };
 
+  private renderLeftArrow = (): React.ReactElement => {
+    if(this.props.arrowLeft) {
+      return (
+        <TouchableWithoutFeedback onPress={this.props.onNavigationLeftPress}>
+          {this.props.arrowLeft}
+        </TouchableWithoutFeedback>
+      )
+    }
```
Comment (artyorsh) [inline src/components/ui/calendar/components/calendarHeader.component.tsx:95]: Can be omitted by passing arrowLeft to Button's accessory prop.
Then there is no need to wrap this component in Touchable for both us and end user.

File: src/components/ui/calendar/components/calendarHeader.component.tsx [hunk 2]
```diff
@@ -82,19 +85,45 @@ export class CalendarHeader extends React.Component<CalendarHeaderProps> {
...
+  };
+
+  private renderRightArrow = (): React.ReactElement => {
+    if(this.props.arrowRight) {
+      return (
+        <TouchableWithoutFeedback onPress={this.props.onNavigationRightPress}>
+          {this.props.arrowRight}
+        </TouchableWithoutFeedback>
+      )
+    }
```
Comment (artyorsh) [inline src/components/ui/calendar/components/calendarHeader.component.tsx:112]: same

File: src/components/ui/calendar/calendar.component.tsx [hunk 3]
```diff
@@ -67,6 +67,14 @@ export type CalendarElement<D = Date> = React.ReactElement<CalendarProps<D>>;
  * @property {(D, NamedStyles) => ReactElement} renderYear - Function component
  * to render instead of default year cell.
  * Called with a date for this cell and styles provided by Eva.
+ * 
+ * @property {(CalendarViewMode) => void} renderArrowLeft - Function component
+ * to render instead of default left arrow.
+ * viewMode returns string with current calendar view ("YEAR", "MONTH", "DATE").
```
Comment (artyorsh) [inline src/components/ui/calendar/calendar.component.tsx:73]: Confusing docs. I'd say we better go with
"Accepts viewMode argument which is..."

File: src/components/ui/calendar/components/calendarHeader.component.tsx [hunk 4]
```diff
@@ -41,6 +42,8 @@ export interface CalendarHeaderProps extends ViewProps {
   onTitlePress?: () => void;
   onNavigationLeftPress?: () => void;
   onNavigationRightPress?: () => void;
+  arrowLeft?: React.ReactElement | null;
+  arrowRight?: React.ReactElement | null;
```
Comment (artyorsh) [inline src/components/ui/calendar/components/calendarHeader.component.tsx:46]: use RenderProp type here from devsupport package

File: src/components/ui/calendar/rangeCalendar.component.tsx [hunk 5]
```diff
@@ -74,6 +74,14 @@ export type RangeCalendarElement<D = Date> = React.ReactElement<RangeCalendarPro
  * to render instead of default year cell.
  * Called with a date for this cell and styles provided by Eva.
  * 
+ * @property {(CalendarViewMode) => void} renderArrowLeft - Function component
+ * to render instead of default left arrow.
+ * viewMode returns string with current calendar view ("YEAR", "MONTH", "DATE").
```
Comment (artyorsh) [inline src/components/ui/calendar/rangeCalendar.component.tsx:79]: Confusing docs

File: src/components/ui/datepicker/datepicker.component.tsx [hunk 6]
```diff
@@ -120,6 +120,17 @@ export type DatepickerElement<D = Date> = React.ReactElement<DatepickerProps<D>>
  * Defaults to *bottom*.
  *
  * @property {StyleProp<ViewStyle>} backdropStyle - Style of backdrop.
+ * 
+ * @property {(CalendarViewMode) => void} renderArrowLeft - Function component
+ * to render instead of default left arrow.
+ * viewMode returns string with current calendar view ("YEAR", "MONTH", "DATE").
```
Comment (artyorsh) [inline src/components/ui/datepicker/datepicker.component.tsx:126]: 📚

**Review by artyorsh — COMMENTED** [review]
It might be OK for now, but not for the future prospective when we could be asked for `renderHeader` prop to customize the whole calendar header.

File: src/components/ui/calendar/calendar.component.tsx (hunk 3 above)
Comment (malashkevich) [reply src/components/ui/calendar/calendar.component.tsx:73]: I would replace `left arrow` with something  more contextual like `previous`

File: src/showcases/components/calendar/calendarCustomArrows.component.tsx [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+      date={date}
+      onSelect={(nextDate) => setDate(nextDate)}
+      renderArrowLeft={renderLeftArrow}
+    />
+  );
+};
+
+const styles = StyleSheet.create({
+  arrow: {
+    display: "flex",
```
Comment (AntonBorovoi) [inline src/showcases/components/calendar/calendarCustomArrows.component.tsx:29]: I guess this one is redundant.
`flex` is default one for `display` property

File: src/showcases/components/datepicker/datepickerCustomArrows.component.tsx [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+      date={date}
+      onSelect={(nextDate) => setDate(nextDate)}
+      renderArrowRight={renderRightArrow}
+    />
+  );
+};
+
+const styles = StyleSheet.create({
+  arrow: {
+    display: "flex",
```
Comment (AntonBorovoi) [inline src/showcases/components/datepicker/datepickerCustomArrows.component.tsx:32]: The same

**Review by sayres — APPROVED** [review]

**Pull request merged**